metadata-ingestion/src/datahub/ingestion/source/sql/stored_procedures/base.py

```diff
@@ -0,0 +1,246 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Callable, Dict, Iterable, Optional
+
+from datahub.emitter.mce_builder import (
+    DEFAULT_ENV,
+    make_data_flow_urn,
+    make_data_job_urn,
+    make_data_platform_urn,
+    make_dataplatform_instance_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.mcp_builder import DatabaseKey, DatahubKey, SchemaKey
+from datahub.ingestion.api.source_helpers import auto_workunit
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.common.subtypes import (
+    FlowContainerSubTypes,
+    JobContainerSubTypes,
+)
+from datahub.ingestion.source.sql.stored_procedures.lineage import parse_procedure_code
+from datahub.metadata.schema_classes import (
+    ContainerClass,
+    DataFlowInfoClass,
+    DataJobInfoClass,
+    DataPlatformInstanceClass,
+    DataTransformClass,
+    DataTransformLogicClass,
+    QueryStatementClass,
+    SubTypesClass,
+)
+from datahub.sql_parsing.schema_resolver import SchemaResolver
+
+
+class ProcedureSignature(DatahubKey):
+    argument_signature: Optional[str] = None
+
+
+@dataclass
+class BaseProcedure:
+    name: str
+    procedure_definition: Optional[str]
+    created: Optional[datetime]
+    last_altered: Optional[datetime]
+    comment: Optional[str]
+    argument_signature: Optional[str]
+    return_type: Optional[str]
+    language: str
+    extra_properties: Optional[Dict[str, str]]
+
+    def get_procedure_identifier(
+        self,
+    ) -> str:
+        argument_signature_suffix = (
+            ProcedureSignature(argument_signature=self.argument_signature).guid()
+            if self.argument_signature
+            else ""
+        )
+        return f"{self.name}_{argument_signature_suffix}"
```

Comment:
this is adding the `_` in any case, right?

what about?

```suggestion
        argument_signature_suffix = (
            ProcedureSignature(argument_signature=self.argument_signature).guid()
            if self.argument_signature
            else None
        )
        return f"{self.name}_{argument_signature_suffix}" if argument_signature_suffix else f"{self.name}"
```

Reply:
Updated